File: typescript/src/openOrders.ts
import { postEndpoint, address } from "./client";

async function main() {
  const result = await postEndpoint("/openOrders", { user: address });

  console.log(`Open orders: ${result.count}`);
  for (const order of result.orders) {
    const side = order.side === "B" ? "BUY" : "SELL";
    const spot = order.isSpot ? " [SPOT]" : "";
    console.log(
      `  ${order.name}${spot} ${side} ${order.sz} @ ${order.limitPx} (OID: ${order.oid})`
    );
  }

  if (result.count > 0) {
    console.log("\nCancel actions by asset:");
    for (const [name, action] of Object.entries<any>(result.cancelActions.byAsset)) {
      console.log(`  ${name}: ${action.cancels.length} order(s) — pass as action to POST /exchange`);
    }

    console.log("\nTo cancel ALL orders, pass cancelActions.all as action to POST /exchange:");
    console.log(JSON.stringify(result.cancelActions.all, null, 2));
  }
}

main();

File: python/open_orders.py
"""View open orders with enriched asset info and pre-built cancel actions."""

import json
from client import post_endpoint, address

result = post_endpoint("/openOrders", {"user": address})

print(f"Open orders: {result['count']}")
for order in result["orders"]:
    side = "BUY" if order["side"] == "B" else "SELL"
    spot = " [SPOT]" if order.get("isSpot") else ""
    print(f"  {order['name']}{spot} {side} {order['sz']} @ {order['limitPx']} (OID: {order['oid']})")

if result["count"] > 0:
    print(f"\nCancel actions by asset:")
    for name, action in result["cancelActions"]["byAsset"].items():
        count = len(action["cancels"])
        print(f"  {name}: {count} order(s) — pass as action to POST /exchange")

    print(f"\nTo cancel ALL orders, pass cancelActions.all as the action to POST /exchange:")
    print(json.dumps(result["cancelActions"]["all"], indent=2))
